main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
well, it is regarding the glibc's behavior.  If you want to handle it differently, I think it should be done outside of MIO -- although I get your point.

BTW, small program very easy to check how `fgets()` behaves:
```c

#include <stdio.h>
#include <string.h>

int main(void)
{
  char buf[32];

  if (fgets(buf, sizeof buf, stdin)) {
    size_t len = strlen(buf);
    size_t i;

    printf("read %lu bytes:\n", len);
    for (i = 0; i < len; i++) {
      printf("  buf[%lu] = ", i);
      switch (buf[i]) {
        case '\r': puts("CR"); break;
        case '\n': puts("LF"); break;
        case '\t': puts("LT"); break;
        case '\v': puts("VT"); break;
        /* ... */
        default:
          putc(buf[i], stdout);
          putc('\n', stdout);
          break;
      }
    }
  }
  return 0;
}
```
and feed it whatever you wanna check.